src/server/pfs/drive/driver.go

```diff
@@ -1071,6 +1072,11 @@ func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint6
 	if fileInfo.FileType == pfs.FileType_FILE_TYPE_NONE {
 		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
 	}
+	// We return NotFound if it's a regular file but its blocks do not match the
+	// given filter shard
+	if fileInfo.FileType == pfs.FileType_FILE_TYPE_REGULAR && len(blockRefs) == 0 && (filterShard != nil && filterShard.BlockNumber > 1) {
+		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
+	}
```

Comment:
So I just tried that and some FUSE tests failed.  I looked into why, and I think it's because when you write a file, the following sequence of FUSE events occur: create -> attr -> write.  When you attr (and therefore `InspectFileUnsafe`), there is no block in the file yet.  So now you get a `file not found` error if you don't check for the existence of a block filter.